### DISCUSSION TITLE
Stream Error On AudioKit

### DISCUSSION THREAD

**Discussion by flashdato** [pschatzmann/arduino-audio-tools#1448] [Q&A] [answered]
my code is working fine on some links and online radios, but when I change it occasionally, it crushes immediately.

Here Is Code 

#include "AudioTools.h"
#include "AudioCodecs/CodecMP3Helix.h"
#include "AudioLibs/AudioBoardStream.h"

#include "BluetoothSerial.h"  //Header File for Serial Bluetooth, will be added by default into Arduino
#include <EEPROM.h>

BluetoothSerial ESP_BT;  //Object for Bluetooth
String buffer_in;
unsigned long previousMillis = 0;
byte val;
int addr = 0;
byte indS = 0;
byte indP = 0;
String stream;
byte len = 0;
String temp;
String temp2;
unsigned int interval = 60000;

const char *urls[] = {
     "http://stream.srg-ssr.ch/m/rsj/mp3_128"
  // "https://dolphin-app-dh42m.ondigitalocean.app/audio/redhat.mp3"
};
char wifi[50];
char password[50];

// Arduino setup
void setup() {

  EEPROM.begin(50);
  Serial.begin(115200);  //Start Serial monitor in 9600
  Serial.println("Bluetooth Device is Ready to Pair");

  Serial.println("Waiting For Wifi Updates 60 seconds");
  ESP_BT.begin("Zgaprebis Yuti");  //Name of your Bluetooth Signal

  while (!check_wifiUpdate() == true) {}
  Serial.println("The Stored Wifi credetial are : ");
  for (int i = 0; i < 50; i++) {
    val = EEPROM.read(i);
    stream += (char)val;
    if ((val == 10) && (indS == 0)) {
      indS = i;
      //Serial.println("indS"+(String)i);
    } else if (val == 10 && indP == 0) {
      indP = i;
      break;
      //Serial.println("indP"+(String)i);
    }
  }
  // Serial.println(stream);
  // Serial.println("Stream Ended");
  temp = stream.substring(0, indS);
  temp = temp.substring(5, indS);
  //ssid2=ssid;
  temp2 = stream.substring(indS + 1, indP);
  temp2 = temp2.substring(5, indP - indS);

  int i = temp.length();
  int j = temp2.length();
  char ssid[i];
  char pass[j];
  temp.toCharArray(ssid, i);
  temp.toCharArray(wifi, i);
  temp2.toCharArray(pass, j);
  temp2.toCharArray(password, j);

  Serial.println("Stored SSID");
  Serial.println(ssid);
  Serial.println("Stored PASS");
  Serial.println(pass);

  AudioLogger::instance().begin(Serial, AudioLogger::Info);

  ESP_BT.end();
}

boolean check_wifiUpdate() {
  unsigned long currentMillis = millis();

  if (currentMillis - previousMillis >= interval) {
    previousMillis = currentMillis;

    Serial.println("60 Seconds Over");

    return true;
  }

  else if (ESP_BT.available())  //Check if we receive anything from Bluetooth
  {
    interval = 50000;
    buffer_in = ESP_BT.readStringUntil('\n');  //Read what we recevive
                                               //Serial.println("Received:"); Serial.println(buffer_in);
    delay(20);

    if (buffer_in.charAt(0) == 'S') {
      for (int i = 0; i < buffer_in.length(); i++) {
        val = (byte)(buffer_in.charAt(i));
        //Serial.println("val "+val);

        EEPROM.write(addr, val);
        //Serial.println(val);
        addr++;
      }
      //Serial.print("New ");
      //Serial.print(buffer_in);
      EEPROM.write(addr, 10);
      addr++;
      EEPROM.commit();
      ESP_BT.println("SSID Stored");
    }

    else if (buffer_in.charAt(0) == 'P') {
      for (int i = 0; i < buffer_in.length(); i++) {
        val = (byte)(buffer_in.charAt(i));
        //Serial.println("val "+val);
        EEPROM.write(addr, val);
        //Serial.println(val);
        addr++;
      }
      //Serial.print("New ");
      //Serial.print(buffer_in);
      EEPROM.write(addr, 10);
      EEPROM.commit();
      ESP_BT.println("Password Stored");
      return true;
    }

    return false;
  } else {
    return false;
  }
}

// Arduino loop
void loop() {
  ICYStream urlStream(wifi, password);
  AudioSourceURL source(urlStream, urls, "audio/mp3");
  AudioBoardStream kit(AudioKitEs8388V1);
  MP3DecoderHelix decoder;
  AudioPlayer player(source, kit, decoder);

  auto cfg = kit.defaultConfig(TX_MODE);
  cfg.sd_active = false;
  kit.begin(cfg);

  urlStream.setTimeout(100);

  player.setVolume(1);
  player.begin();

  delay(1000);

  while (1) {
    player.copy();
    kit.processActions();
  }
}

![image](https://github.com/pschatzmann/arduino-audio-tools/assets/63307225/282365bb-7d3a-4a13-8b43-4779e2521b28)

as u can see, the first links work just fine but when I turn it into the second it crushes. in the last discussion, I had the same issue and it was solved I think the library needs a little update @pschatzmann 

here is error message 

![image](https://github.com/pschatzmann/arduino-audio-tools/assets/63307225/9d029b85-b295-4f55-b378-6c705a792a37)

**Comment by pschatzmann** [comment]
Just look at your error message: you seem to run out of stack, so try to increase it...

**Reply by flashdato** [reply]
how exactly?

**Reply by pschatzmann** [reply]
e.g. #define ARDUINO_LOOP_STACK_SIZE (30 * 1024)

**Reply by flashdato** [reply]
It does not work I tried to increase it to 200. it has the same error

**Reply by pschatzmann** [reply, marked as answer]
Then you have some error in your code and you create some buffer overflow  ...
If you think that the issue is in my code, please provide some small test sketch that demonstrates the issue.
